client/apps/edit/components/edit_container.jsx

```diff
@@ -1,39 +1,70 @@
 import PropTypes from 'prop-types'
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
-import { changeSavedStatus, saveArticle } from 'client/actions/editActions'
-
+import {
+  changeSavedStatus,
+  saveArticle,
+  toggleSpinner
+} from 'client/actions/editActions'
 import { ErrorBoundary } from 'client/components/error/error_boundary'
 import { EditAdmin } from './admin/index.jsx'
 import { EditContent } from './content/index.jsx'
 import { EditDisplay } from './display/index.jsx'
 import EditHeader from './header/index.jsx'
 import EditError from './error/index.jsx'
 
-class EditContainer extends Component {
+export class EditContainer extends Component {
   static propTypes = {
     activeView: PropTypes.string,
     article: PropTypes.object,
     changeSavedStatusAction: PropTypes.func,
     channel: PropTypes.object,
     error: PropTypes.object,
+    isSaved: PropTypes.bool,
     saveArticleAction: PropTypes.func,
-    user: PropTypes.object
+    toggleSpinnerAction: PropTypes.func
   }
 
   constructor (props) {
     super(props)
 
-    this.state = {
-      lastUpdated: null
-    }
+    this.beforeUnload = this.beforeUnload.bind(this)
+    this.setupBeforeUnload()
 
     props.article.sections.on(
       'change add remove reset',
       () => this.maybeSaveArticle()
     )
   }
 
+  componentDidMount = () => {
+    this.props.toggleSpinnerAction(false)
+  }
+
+  setupBeforeUnload = () => {
+    const { article } = this.props
+
+    if (article.get('published')) {
+      article.on(
```

Comment:
This looks like it will add a new event listener on every `change ` event for the current article model, wouldn't `.once` be preferable here?

Reply:
Ah yes you are right -- once is definitely preferable here.

Reply:
Actually, why not use the `maybeSaveArticle` or `onChange` event already in the class to check if we should add the listener instead of introducing a new backbone event listener?

https://github.com/artsy/positron/pull/1567/files#diff-b18066acb67f8f3287b04ae47476bc57R83